scripts/build/build.js

```diff
@@ -1,108 +1,230 @@
+const util = require("./util");
+
+const EXTERNALS = [
+  "assert",
+  "buffer",
+  "constants",
+  "crypto",
+  "events",
+  "fs",
+  "module",
+  "os",
+  "path",
+  "stream",
+  "url",
+  "util",
+  "readline",
+
+  // See comment in jest.config.js
+  "graceful-fs"
+];
+
+function getBabelConfig(bundle) {
+  const config = {
+    babelrc: false,
+    presets: [],
+    plugins: ["external-helpers"]
+  };
+  if (bundle.type === "core") {
+    config.plugins.push(require("./babel-plugins/transform-eval-require"));
+  }
+  if (bundle.transpile) {
+    config.presets.push(["es2015", { modules: false }]);
+  }
+  return config;
+}
 
-const rootDir = path.join(__dirname, "..", "..");
+function getRollupConfig(bundle) {
+  const relative = fp => `./${path.basename(fp).replace(/\.js$/, "")}`;
+  const paths = (bundle.external || []).reduce(
+    (paths, filepath) =>
+      Object.assign(paths, { [filepath]: relative(filepath) }),
+    { "graceful-fs": "fs" }
+  );
 
-process.env.PATH += path.delimiter + path.join(rootDir, "node_modules", ".bin");
+  const config = {
+    entry: bundle.input,
+    paths,
+
+    onwarn(warning) {
+      // We use `eval` to force rollup to code split and
+      // ignore warnings from node_modules
+      if (
+        warning.code === "EVAL" ||
+        (warning.code === "CIRCULAR_DEPENDENCY" &&
+          warning.importer.startsWith("node_modules"))
+      ) {
+        return;
+      }
+
+      // web bundle can't have external requires
+      if (
+        warning.code === "UNRESOLVED_IMPORT" &&
+        bundle.target === "universal"
+      ) {
+        throw new Error(
+          `Unresolved dependency in universal bundle: ${warning.source}`
+        );
+      }
+
+      console.log(warning);
+    }
+  };
+
+  if (bundle.target === "node") {
+    Object.assign(config, {
+      plugins: [
+        replace(
+          Object.assign(
+            { "process.env.NODE_ENV": "'production'" },
+            bundle.replace
+          )
+        ),
+        bundle.executable ? executable() : {},
+        json(),
+
+        resolve({
+          extensions: [".js", ".json"],
+          preferBuiltins: true
+        }),
+
+        commonjs(bundle.commonjs || {}),
+        babel(getBabelConfig(bundle)),
+        bundle.minify ? uglify() : {}
+      ],
+      external: EXTERNALS.concat(bundle.external)
+    });
+  } else if (bundle.target === "universal") {
+    Object.assign(config, {
+      plugins: [
+        replace(
+          Object.assign(
+            {
+              "process.env.NODE_ENV": "'production'",
+              "process.env.PRETTIER_DEBUG": "global.PRETTIER_DEBUG"
+            },
+            bundle.replace
+          )
+        ),
+        json(),
+        nativeShims(path.resolve(__dirname, "shims")),
+        resolve({
+          extensions: [".js", ".json"],
+          preferBuiltins: false
+        }),
+        commonjs(bundle.commonjs || {}),
+        nodeGlobals()
+        // babel(),
+        // uglify()
```

Comment:
Why are these commented out? Should we remove these lines, or do they need to be uncommented?

Reply:
The `babel()` is just a reminder to myself that we should probably just use `babel-preset-env` and include `node 4` in the targets

I commented `uglify()` because it takes too long to run, but I wasn't supposed to commit that